### PR TITLE
[MetricsAdvisor] Implemented BoundaryMeasureType

### PR DESCRIPTION
**Pull request by kinelski**
Part of https://github.com/azure/azure-sdk-for-net/issues/21670.

Adding new type and new property.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs:21]: Asked the service how many data points are used in the calculation of this mean. I'll wait for their response before merging this PR.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:209]: For most tests, we'll leave the `MeasureType` property untouched. We'll just check that its default value (`BoundaryMeasureType.Value`) has been set.

We have one "Create-operation" test (this one) that sets the property to `Mean`, and one "Update-operation" test that updates the property to `Mean`.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs:16]: nit: it could be me and English but feels like it should be `of a metric` or `of the metric`

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs
Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs:21]: Does the user set the time series somewhere? or is this part of the answer you are waiting for

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/MetricBoundaryCondition.cs
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/MetricBoundaryCondition.cs:62]: nit: ... which measure type should

**Review by maririos — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs:16]: You're right. Sounds awkward. Updated to "of the metric".

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/BoundaryMeasureType.cs:21]: The time series are defined by the user-provided data source that is set during data feed creation.

A time series represents a value that varies over time. It could be, for instance, the amount of electrical energy consumed by a factory in a single day, or the total cost a company had in the period of one month. The value being measured (the "metric") is registered periodically (daily, monthly, etc), and monitored by the service.

An example of time series (plotted as a graph over time):
image

These red spots are anomalies detected by the service. The service, however, will not notify the customer every single time an anomaly is detected. Users need to set an Alert Configuration to be notified. Every time an anomaly is detected in a time series, it will go through this configuration to check if it's an anomaly worth being reported. This new type, `BoundaryMeasureType`, is used as part of this Alert Configuration. Every time we mention a "time series" in this context, it should be interpreted as "the time series in which the current anomaly was detected".

**Pull request merged**
